Sources/Models/FunctionCall.swift

```diff
@@ -7,12 +7,263 @@
 
 import Foundation
 
-public struct FunctionCall {
-    enum CodingKeys: CodingKey {
+// MARK: - Shared Helper
+
+/// Helper struct for encoding/decoding Any values used across function/tool call parsing
+public struct AnyCodable: Codable {
+    public let value: Any
+    
+    public init(_ value: Any) {
+        self.value = value
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        
+        if let stringValue = try? container.decode(String.self) {
+            value = stringValue
+        } else if let intValue = try? container.decode(Int.self) {
+            value = intValue
+        } else if let doubleValue = try? container.decode(Double.self) {
+            value = doubleValue
+        } else if let boolValue = try? container.decode(Bool.self) {
+            value = boolValue
+        } else if let arrayValue = try? container.decode([AnyCodable].self) {
+            value = arrayValue.map { $0.value }
+        } else if let dictionaryValue = try? container.decode([String: AnyCodable].self) {
+            value = dictionaryValue.mapValues { $0.value }
+        } else if container.decodeNil() {
+            value = NSNull()
+        } else {
+            throw DecodingError.typeMismatch(Any.self, DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "Unsupported type"))
+        }
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        
+        switch value {
+        case let stringValue as String:
+            try container.encode(stringValue)
+        case let intValue as Int:
+            try container.encode(intValue)
+        case let doubleValue as Double:
+            try container.encode(doubleValue)
+        case let boolValue as Bool:
+            try container.encode(boolValue)
+        case let arrayValue as [Any]:
+            let anyCodableArray = arrayValue.map { AnyCodable($0) }
+            try container.encode(anyCodableArray)
+        case let dictionaryValue as [String: Any]:
+            let anyCodableDictionary = dictionaryValue.mapValues { AnyCodable($0) }
+            try container.encode(anyCodableDictionary)
+        case is NSNull:
+            try container.encodeNil()
+        default:
+            throw EncodingError.invalidValue(value, EncodingError.Context(codingPath: encoder.codingPath, debugDescription: "Unsupported type"))
+        }
+    }
+}
+
+// MARK: - Legacy Format Model
+
+/// LEGACY: Represents a function call from old "function-call" messages
+/// Kept for backward compatibility with older Vapi versions
```

Comment:
❓ Does this mean that we may use this SDK with an old version of the Vapi backend that still uses function calls? If not, I wonder if we can get rid of function calls from the SDK code.